tests/test_cli.py

```diff
@@ -1,198 +1,297 @@
+
+    def test_non_existent_directory(self, tmp_path: Path):
+        """Test that providing a non-existing directory to write to is OK."""
+        out = tmp_path / "parent" / "config.toml"
+        app(f"template create --template simple --out {out}")
+        assert out.exists()
+
+
+class TestRunSetup:
+    """Tests of the _run_setup function."""
+
+    def setup_class(self):
+        """Make a default temp dir, and store a simple config TOML."""
+        self.tmpdir = Path(tempfile.mkdtemp())
+
+        self.simple = self.tmpdir / "simple.toml"
+
+        # Create a full template in tmpdir
+        app(f"template create --template simple --out {self.simple}")
+
+    def test_unmodified_paramfile(self, capsys):
+        """Test that running with an unmodified --param-file doesn't write the fullspec."""
+        runp = RunParams(
+            param_selection=ParameterSelection(param_file=self.simple),
+            cachedir=self.tmpdir,
+        )
+        params = Parameters()  # don't modify the input
+        _run_setup(runp, params)
+        out = capsys.readouterr().out
+        assert "Wrote full configuration" not in out
+
+    def test_unmodified_template(self, capsys):
+        """Test that an unmodified --template does write a simple fullspec TOML."""
+        runp = RunParams(
+            param_selection=ParameterSelection(template="simple"), cachedir=self.tmpdir
+        )
+        params = Parameters()  # don't modify the input
+        _run_setup(runp, params)
+        out = capsys.readouterr().out
+        assert "Wrote full configuration" in out
+        assert "simple.toml" in out
+
+    def test_explicit_outcfg(self, capsys):
+        """Test that directly modifying params and passing an explicit file works."""
+        outcfg = self.tmpdir / "custom-name.toml"
+        runp = RunParams(
+            param_selection=ParameterSelection(template="simple"),
+            outcfg=outcfg,
+            cachedir=self.tmpdir,
+        )
+        params = Parameters(simulation_options=cli._SimulationOptions(HII_DIM=37))
+
+        _run_setup(runp, params)
+        out = capsys.readouterr().out
+        assert "Wrote full configuration" in out
+        assert f"{outcfg}" in out
+
+        _run_setup(runp, Parameters())
+        out = capsys.readouterr().out
+        assert "Wrote full configuration" in out
+        assert f"{outcfg}" not in out
+        assert "simple.toml" in out
+
+    def test_unknown_name(self, capsys):
+        """Test that modifying params without an explicit file creates a random file."""
+        runp = RunParams(
+            param_selection=ParameterSelection(template="simple"), cachedir=self.tmpdir
+        )
+        params = Parameters(simulation_options=cli._SimulationOptions(HII_DIM=37))
+
+        _run_setup(runp, params)
+        out = capsys.readouterr().out
+        assert "Wrote full configuration" in out
+        assert "simple.toml" not in out  # got a random uuid
```

Comment:
Just so I understand the rationale here: since we already tested the template writing setup earlier it's fine to take the output log lines as evidence that the actual writing was done correctly? I'm fine with this I just wanted to make sure it was intentional